models/classes/GenerisTreeFactory.php

```diff
@@ -364,7 +364,7 @@ private function mustFilterTranslations(core_kernel_classes_Class $class): bool
         $mainClass = array_pop($parentClassIds);
 
         return in_array(
-            $mainClass,
+            $mainClass ?? $class->getUri(),
```

Comment:
We can remove lines `363-364` and use the following code here:
```suggestion
            $class->getRootId() ?? $class->getUri(),
```